src/lib.rs

```diff
@@ -30,6 +30,31 @@ mod unix;
 #[cfg(unix)]
 use crate::unix as sys;
 
+#[cfg(all(unix, feature = "multilock"))]
+mod unix_fileid;
+#[cfg(all(unix, feature = "multilock"))]
+use unix_fileid as fileid;
+#[cfg(not(all(unix, feature = "multilock")))]
+mod nil_fileid;
+#[cfg(not(all(unix, feature = "multilock")))]
+use nil_fileid as fileid;
+
+/// Enumeration used to declare whether FsLock instances opened with the same
+/// file, by the same process, are exclusive.
+#[derive(Debug,Copy,Clone)]
+#[non_exhaustive]
+enum Exclusive {
```

Comment:
I think this name represents the semantics better: this is the ways which you can be exclusive (either per fd or os-dependent), so I'd add the suffix -ity.
```suggestion
enum Exclusivity {
```